fs/expose/dokan/__init__.py

```diff
@@ -826,43 +910,37 @@ def _filetime2datetime(ftime):
         return DATETIME_ZERO
     return _timestamp2datetime(_filetime2timestamp(ftime))
 
+
 def _datetime2filetime(dtime):
     """Convert a FILETIME struct info datetime.datetime object."""
     if dtime is None:
-        return libdokan.FILETIME(0,0)
+        return libdokan.FILETIME(0, 0)
     if dtime == DATETIME_ZERO:
-        return libdokan.FILETIME(0,0)
+        return libdokan.FILETIME(0, 0)
     return _timestamp2filetime(_datetime2timestamp(dtime))
 
 
 def _errno2syserrcode(eno):
     """Convert an errno into a win32 system error code."""
     if eno == errno.EEXIST:
-        return ERROR_FILE_EXISTS
+        return STATUS_OBJECT_NAME_COLLISION
     if eno == errno.ENOTEMPTY:
-        return ERROR_DIR_NOT_EMPTY
+        return STATUS_DIRECTORY_NOT_EMPTY
     if eno == errno.ENOSYS:
-        return ERROR_NOT_SUPPORTED
+        return STATUS_NOT_SUPPORTED
     if eno == errno.EACCES:
-        return ERROR_ACCESS_DENIED
+        return STATUS_ACCESS_DENIED
     return eno
```

Comment:
I think, it's better to return here some STATUS_* constant to clearly indicate error. Dokan/Windows tend to ignore invalid status codes, as far as I understand, and errno values is mostly invalid as status codes.

In my experiments Dokan could irreversibly hang both the FS process and explorer.exe because of such errors, to the point of inability of reboot Windows.